sina: new package

estebanpauli opened the pull request.

Commit:
Added spackage to build Sina (https://github.com/LLNL/Sina).

tgamblin commented on var/spack/repos/builtin/packages/sina/package.py:
```diff
@@ -0,0 +1,101 @@
+    homepage = 'https://github.com/LLNL/Sina'
+    url = 'https://github.com/LLNL/Sina/releases/download/v1.10.0/sina-cpp-1.10.0.tar.gz'
+
+    version('develop', git='git@github.com:LLNL/Sina.git',
```
don't use the `git` protocol URL here; it won't work unless you have ssh set up.  Use the https URL.

estebanpauli replied:
Turns out we don't even have a develop branch on github :). Most of our work is done internally and then pushed when a release is ready. I will just remove this.

tgamblin commented on var/spack/repos/builtin/packages/sina/package.py:
```diff
@@ -0,0 +1,101 @@
+            description='Create interface for calling Sina through Adiak')
+    # Availability of the adiak version with optional mpi varies
+    variant('adiak-no-mpi', default=False,
+            description='Create interface for calling Sina through Adiak')
```
It looks like this juts controls a variant on a dependency, but there's no need for a separate option for that.  You can just do:

```console
$ spack install sina ^adiak~mpi
```

and then you don't need the same piece of information represented twice.

estebanpauli replied:
I did not realize I could do this. I will clean this up along with changing the conduit line below.

tgamblin commented on var/spack/repos/builtin/packages/sina/package.py:
```diff
@@ -0,0 +1,101 @@
+    depends_on('cmake@3.8.0:', type='build')
+    depends_on('adiak', when='+adiak')
+    depends_on('mpi', when='+adiak')
+    depends_on('adiak ~mpi', when='+adiak-no-mpi')
```
can delete this if you agree w/my comment above.

tgamblin commented on var/spack/repos/builtin/packages/sina/package.py:
```diff
@@ -0,0 +1,101 @@
+    depends_on('mpi', when='+adiak')
+    depends_on('adiak ~mpi', when='+adiak-no-mpi')
+    depends_on('doxygen', type='build', when='+docs')
+    depends_on('conduit ~python ~mpi ~hdf5 ~silo ~shared ~test ~hdf5_compat')
```
this means that nobody can use `sina` with conduit with `python`, etc. enabled.  These types of restrictions really belong in the environment you're building your app in, not in `sina`.

tgamblin commented on var/spack/repos/builtin/packages/sina/package.py:
```diff
@@ -0,0 +1,101 @@
+    def configure_args(self):
+        spec = self.spec if self.spec is not None else ''
+        return [
+            '-DSINA_BUILD_ADIAK_BINDINGS={0}'.format('YES' if '+adiak' in spec or '+adiak-no-mpi' in spec else 'NO'),
```
This can actually just be:

```python
    self.define_from_variant('SINA_BUILD_ADIAK_BINDINGS', 'adiak')
```

same for ones below

tgamblin commented on var/spack/repos/builtin/packages/sina/package.py:
```diff
@@ -0,0 +1,101 @@
+
+        # Try to find the common prefix of the TPL directory, including the compiler
+        # If found, we will use this in the TPL paths
+        compiler_str = str(self.spec.compiler).replace('@','-')
```
```python
self.spec.format("{compiler}-{compiler.version}")
```

tgamblin commented on var/spack/repos/builtin/packages/sina/package.py:
```diff
@@ -0,0 +1,101 @@
+        if len(prefix_paths) == 2:
+            tpl_root = os.path.join( prefix_paths[0], compiler_str )
+            path_replacements[tpl_root] = '${TPL_ROOT}'
+            entries.append(cmake_cache_path('TPL_ROOT', tpl_root))
```
Is this pointing at some LLNL-specific TPL root?  If so this doesn't belong in an upstream package -- you should use externals for these roots...

estebanpauli replied:
Hmm.... I don't remember how this works.

We are using BLT and uberenv to build our library. This file is adapted from there. I'll have to figure out how this all works again.

I think though that this is just creating host-config files, and defining a variable named `TPL_ROOT` in them to be able to figure out where the third-party libraries are.

If this is the case (using this to create host-configs), should I just drop this from this package file?

estebanpauli replied:
I did some looking back, and this is to have a variable named 'TPL_ROOT' in the host-config files. Everything that is relative to this is changed to use this variable. This make it easier to spot the commonalities. For the public build, this is not important to have in the cache file. I will remove it.

tgamblin commented on var/spack/repos/builtin/packages/sina/package.py:
```diff
@@ -0,0 +1,101 @@
+        entries.append(cmake_cache_path(
+            'Conduit_DIR', '{}/lib/cmake/conduit'.format(conduit_dir)))
+
+        use_adiak = '+adiak' in self.spec or '+adiak-no-mpi' in self.spec
```
Even if you end up keeping both of these variants, you can check whether you depend on `adiak` or not with:

```python
'adiak' in spec
```

or

```python
spec.satisfies('^adiak')
```

Commit:
Improvements to sina/package.py

Made numerous simplifications and improvements to sina/package.py
based on PR feedback.

Commit:
Added licence info

Commit:
Added maintainers

estebanpauli commented:
I have addressed all requested changes and this is now ready for the next review.

tgamblin commented on var/spack/repos/builtin/packages/sina/package.py:
```diff
@@ -0,0 +1,78 @@
+    homepage = 'https://github.com/LLNL/Sina'
+    url = 'https://github.com/LLNL/Sina/releases/download/v1.10.0/sina-cpp-1.10.0.tar.gz'
+
+    maintainers = [
+        'pauli2@llnl.gov',
+        'haluska2@llnl.gov',
+        'murray55@llnl.gov',
+    ]
```
These should be GitHub usernames. Spackbot will ping these people when updates to the package are submitted in PRs.

```suggestion
    maintainers = ["estebanpauli", "HaluskaR"]
```

tgamblin replied:
I'm not sure what murray55's GitHub username is or if he has one so if you like this you can just commit the suggestion; if not push something and we'll get this merged.

tgamblin reviewed (changes requested):
LGTM -- one fix on the `maintainers` list.

Commit:
Changed maintainers to be Github IDs.
